experimental/segmenter/src/lib.rs

```diff
@@ -152,6 +153,8 @@ mod line;
 mod sentence;
 mod word;
 
+// icu_datagen uses provider, but we don't want to expose this implementation detail to the users.
+#[doc(hidden)]
```

Comment:
Most of the `provider` module is already documented, and we generally keep it public in other crates, since it is effectively part of our public API. I think you can document or suppress docs on particular structs/fields.

Reply:
Will revert this hunk, and add more docs.